src/battle/action_cmd/jump.c

```diff
@@ -2,12 +2,203 @@
+    gBattleStatus.unk_434 = &D_802941C0;
+    gBattleStatus.unk_86 = 0;
 
-INCLUDE_ASM(s32, "battle/action_cmd/jump", func_802A9590_421F80);
+    if (gBattleStatus.unk_83 == 0) {
+        gBattleStatus.actionSuccess = 0;
+        return ApiStatus_DONE2;
+    }
+    
+    func_80268858();
+    actionCommandStatus->actionCommandID = 1;
+    actionCommandStatus->hudElementX = -0x30;
+    actionCommandStatus->state = 0;
+    actionCommandStatus->unk_60 = 0;
+    actionCommandStatus->hudElementY = 0x50;
+
+    hudElement = create_hud_element(&D_80108B28);
+    actionCommandStatus->hudElements[0] = hudElement;
+    set_hud_element_flags(hudElement, 0x82);
+    set_hud_element_render_pos(hudElement, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hudElement, 0);
+    set_hud_element_alpha(hudElement, 0xFF);
+
+    hudElement = create_hud_element(&D_80292A2C);
+    actionCommandStatus->hudElements[1] = hudElement;
+    set_hud_element_flags(hudElement, 0x82);
+    set_hud_element_render_pos(hudElement, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(hudElement, 0);
+    set_hud_element_alpha(hudElement, 0xFF);
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A9120_421B10(ScriptInstance* script, s32 isInitialCall) {
+    HudElement* hudElement;
+
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    BattleStatus* battleStatus = &gBattleStatus;
+    Bytecode* args = script->ptrReadPos;
+
+    if (battleStatus->unk_83 == 0) {
+        battleStatus->actionSuccess = 0;
+        return ApiStatus_DONE2;
+    } else {
+        func_80268858();
+        actionCommandStatus->unk_4E = get_variable(script, *args++);
+        actionCommandStatus->unk_50 = get_variable(script, *args++);
+        actionCommandStatus->unk_50 = func_80268224(actionCommandStatus->unk_50);
+        actionCommandStatus->unk_60 = 0;
+        battleStatus->actionSuccess = 0;
+
+        hudElement = actionCommandStatus->hudElements[0];
+        actionCommandStatus->hudElementX = 0x32;
+        battleStatus->flags1 &= ~0x8000;
+        battleStatus->flags1 &= ~0x2000;
+        set_hud_element_render_pos(hudElement, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+        if (actionCommandStatus->unk_61 != 0) {
+            clear_hud_element_flags(hudElement, 2);
+        }
+
+        actionCommandStatus->state = 10;
+        func_80269118();
+        btl_set_popup_duration(0xA);
+        return ApiStatus_DONE2;
+    }
+}
+
+void N(update)(void) {
+    HudElement* hudElement;
+    s32 temp_s0_3;
+
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    BattleStatus* battleStatus = &gBattleStatus;
+
+    switch (actionCommandStatus->state) {
+    case 0:
+        if (battleStatus->unk_83 == 2) {
+            btl_set_popup_duration(0x63);
+        }
+        actionCommandStatus->state = 1;
+        return;
```

Comment:
in a switch, `break` is a little more preferred than returning. THere are some parts of this where the control flow can be cleaned up a little, like the end of case 1. You can just `break` instead of falling through to the default case with the return. you also don't need the default case